zebra-consensus/src/transaction.rs

```diff
@@ -645,15 +645,18 @@ where
     }
 
     /// Attempts to find a transaction in the mempool by its transaction hash and checks
-    /// that all of its dependencies are available in the block.
+    /// that all of its dependencies are available in the block or in the state.  Waits
+    /// for UTXOs being spent by the given transaction to arrive in the state if they're
+    /// not found elsewhere.
     ///
     /// Returns [`Some(Ok(VerifiedUnminedTx))`](VerifiedUnminedTx) if successful,
     /// None if the transaction id was not found in the mempool,
     /// or `Some(Err(TransparentInputNotFound))` if the transaction was found, but some of its
     /// dependencies are missing in the block.
-    async fn try_find_verified_unmined_tx(
+    async fn find_verified_unmined_tx(
         req: &Request,
         mempool: Option<Timeout<Mempool>>,
+        state: Timeout<ZS>,
     ) -> Option<Result<VerifiedUnminedTx, TransactionError>> {
         if req.is_mempool() || req.transaction().is_coinbase() {
```

Comment:
###### Optional

The `Request::transaction` fn clones the inner tx, so it's a bit more optimal to call it only once. This suggestion requires the one above.

```suggestion
        let tx = req.transaction();

        if req.is_mempool() || tx.is_coinbase() {
```

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/9118/commits/97a1c4052863fb6b90ca1b54a7aa8af2355a74ae.